### PR TITLE
new: introduce plugin loader code

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area plugin-sdk

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

After the changes in libs of https://github.com/falcosecurity/libs/pull/392 (~yet to get merged~), the plugin loading code is isolated in a self-contained package and implements a C api. As such, it is possible to develop loaders in other programming languages by reusing the libs one, which makes it compliant with the targeted libs version out of the box.

This first version of the loader is capable of loading plugins, validating them against the currently-supported plugin API version, and reading all their static/descriptive information (name, version, etc...). The loader is also capable of initializing the plugin in order to read initialization-dependent information such as the list of suggested open params.

**Special notes for your reviewer**:

Blocked and WIP until merging https://github.com/falcosecurity/plugin-sdk-go/pull/63.

Since this PR is already big, unit tests and examples will be added in a future PR. Support for more operations on the loaded plugins will be considered in the future.

Most of the changes are related to pulling `plugin_loader.h/.c` from libs. The actual loader code is ~400 LOC and resides in pkg/loader/loader.go.

**Does this PR introduce a user-facing change?**:

```release-note
new: introduce plugin loader code
```

### CODE REVIEW

**Comment by jasondellaluce** [comment]
I splitted the libs version bump contribution to its own PR https://github.com/falcosecurity/plugin-sdk-go/pull/63, and rebase this one on top of that. This PR is blocked until #63 gets merged.

/hold

**Review by leogr — APPROVED** [review]
This is great!
I can't wait to use this loader. :star_struck:

**Pull request merged**
